client/config/config_test.go

```diff
@@ -0,0 +1,104 @@
+package config_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/config"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/x/staking/client/cli"
+)
+
+const (
+	nodeEnv   = "NODE"
+	testNode1 = "http://localhost:1"
+	testNode2 = "http://localhost:2"
+)
+
+func initContext(t *testing.T) (context.Context, func()) {
+	home := t.TempDir()
+
+	clientCtx := client.Context{}.
+		WithHomeDir(home).
+		WithViper()
+
+	clientCtx.Viper.BindEnv(nodeEnv)
+
+	clientCtx, err := config.ReadFromClientConfig(clientCtx)
+	require.NoError(t, err)
+
+	ctx := context.WithValue(context.Background(), client.ClientContextKey, &clientCtx)
+	return ctx, func() { _ = os.RemoveAll(home) }
+}
+
+func TestConfigCmd(t *testing.T) {
+	os.Setenv(nodeEnv, testNode1)
+	ctx, cleanup := initContext(t)
+	defer func() {
+		os.Unsetenv(nodeEnv)
+		cleanup()
+	}()
+
+	cmd := config.Cmd()
+	// NODE=http://localhost:1 ./build/simd config node http://localhost:2
+	cmd.SetArgs([]string{"node", testNode2})
+	require.NoError(t, cmd.ExecuteContext(ctx))
```

Comment:
I believe you can use `testutil/cli.ExecTestCLICmd`, which might save a couple of lines